fe/fe-core/src/main/java/com/starrocks/transaction/TransactionStateBatch.java

```diff
@@ -147,6 +147,18 @@ public List<TransactionState> getTransactionStates() {
         return transactionStates;
     }
 
+    public void writeLock() {
```

Comment:
need to sort transactionStates first to avoid dead lock?

Reply:
This list is already in order and will be sorted according to transaction dependencies.